cinema/views.py

```diff
@@ -0,0 +1,41 @@
+from django.http import HttpRequest
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+
+from .models import Movie
+from .serializers import MovieSerializer
+
+
+@api_view(["GET", "POST"])
+def movies(request: HttpRequest) -> Response:
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "POST":
+        serializer = MovieSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request: HttpRequest, pk: int) -> Response:
+    if request.method == "GET":
+        movie = get_object_or_404(Movie, pk=pk)
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "PUT":
+        movie = get_object_or_404(Movie, pk=pk)
+        serializer = MovieSerializer(movie, request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "DELETE":
+        get_object_or_404(Movie, pk=pk).delete()
```

Comment:
since only these 3 methods are accepted here you don't need last if statement + you can move `movie = get_object_or_404(Movie, pk=pk)` before all if's and then reuse variable in each conditional